32_awk_lab/awklab.txt

```diff
@@ -1,11 +1,12 @@
 1. Write an awk one-liner that prints the file "shells" with a blank line after each line
-Answer: 
+Answer: awk "{ print $0, "\n"}' /etc/shells
 
 
 2. Write an awk one-liner that prints the file "shells" with a ">" sign at the beginning of each line
-Answer: 
+Answer: awk '{ print ">"$0}' /etc/shells
 
 3. Write an awk one-liner to swap the first and last words in the output of "help cd"
+awk '{temp = $1, $1 = $NF, $NF = temp}' /bin/cd
```

Comment:
3 things to note here:
* you need to replac ```,``` with ```;```
* you need to pipe the output of the command ```help cd```
* and you need to add the ```print``` action.

something like this should work:
``` help cd | awk '{temp=$1; $1=$NF; $NF=temp}; {print}'```

Note that I added a print action at the end, ```awk``` will look for ```pattern {action}``` pairs, if pattern is omitted the action that flows  will match anything. If  {action} is omitted it will default to {print}, so may times you will see people write it like this:
```help cd | awk '{temp=$1; $1=$NF; $NF=temp};1'```

```1``` is a pattern and the missing action will default to ```{print}```